src/python/ensembl/hive/HiveRESTClient.py

```diff
@@ -92,18 +91,13 @@ def fetch_input(self):
         Return response received.
         """
         with self._session_scope() as http:
-            self.response = http.request(method=self.param('method'),
-                                         url=self.param('endpoint'),
+            response = http.request(self.param_required('method'),
```

Comment:
```suggestion
            response = http.request(self.param('method'),
```
I may misunderstand the param_required thing, but since this value is defaulted by "get", why make it `required`?

Reply:
There are two reasons
1. `param_required` checks that the value is not _None_. Even if the default value is `get` (in the Runnable's `param_defaults`, the parameter could still be set to _None_ somewhere else (Runnable's `param_defaults` is at the bottom of the priority list, cf https://ensembl-hive.readthedocs.io/en/master/creating_pipelines/parameters.html#parameter-sources )
2. A subclass may overwrite `param_defaults` and neither call `super().param_defaults()` nor provide a default value for `method`

I would also say it's about conveying the intent: `method` is a mandatory parameter that really has to be set.

Reply:
Ok, thank you @muffato , I got it now! That's fine with the required, didn't know about the None thing. This makes sense then.